### PR TITLE
llvm: Symlink OpenMP header/libs to separate directory

### PR DESCRIPTION
**Pull request by ts826848**
This allows other formulae to use omp.h and libomp.dylib without pulling
in other libraries installed with LLVM that conflict with system ones
(e.g. libc++.dylib)

- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

For [PR #2234 at Homebrew/brew](https://github.com/Homebrew/brew/pull/2234).

### CODE REVIEW

File: Formula/llvm.rb [hunk 1]
```diff
@@ -272,6 +272,13 @@ def install
     # install llvm python bindings
     (lib/"python2.7/site-packages").install buildpath/"bindings/python/llvm"
     (lib/"python2.7/site-packages").install buildpath/"tools/clang/bindings/python/clang"
+
+    # Install symlinks for omp.h/libomp so they can be used in other formulae
+    mkdir lib/"libomp"
+    ["libgomp.dylib", "libiomp5.dylib", "libomp.dylib"].each { |libomp_dylib|
+      (lib/"libomp").install_symlink lib/"#{libomp_dylib}"
```
Comment (DomT4) [inline Formula/llvm.rb]: The use of `install_symlink` here should make the `mkdir` above redundant.

File: Formula/llvm.rb [hunk 2]
```diff
@@ -272,6 +272,13 @@ def install
     # install llvm python bindings
     (lib/"python2.7/site-packages").install buildpath/"bindings/python/llvm"
     (lib/"python2.7/site-packages").install buildpath/"tools/clang/bindings/python/clang"
+
+    # Install symlinks for omp.h/libomp so they can be used in other formulae
+    mkdir lib/"libomp"
+    ["libgomp.dylib", "libiomp5.dylib", "libomp.dylib"].each { |libomp_dylib|
```
Comment (MikeMcQuaid) [inline Formula/llvm.rb]: Check `brew style`: https://bot.brew.sh/job/Homebrew%20Core/17250/version=sierra/testReport/junit/brew-test-bot/sierra/style_llvm/

I'd also use just `|dylib|` here.

File: Formula/llvm.rb [hunk 3]
```diff
@@ -272,6 +272,13 @@ def install
     # install llvm python bindings
     (lib/"python2.7/site-packages").install buildpath/"bindings/python/llvm"
     (lib/"python2.7/site-packages").install buildpath/"tools/clang/bindings/python/clang"
+
+    # Install symlinks for omp.h/libomp so they can be used in other formulae
```
Comment (MikeMcQuaid) [inline Formula/llvm.rb]: `# Install symlinks for OpenMP header and libraries so they can be used by superenv`

**Comment by ilovezfs** [comment]
I'm not sure I understand why this isn't an upstream bug if we think this is necessary.

File: Formula/llvm.rb (hunk 2 above)
Comment (aw1621107) [inline Formula/llvm.rb]: Should `lib/"#{dylib}"` be replaced with `lib/dylib.to_s`, or does `dylib.to_s` need parentheses?

**Comment by aw1621107** [comment]
@ilovezfs Sorry, I'm not quite sure what you're getting at. Could you elaborate?

File: Formula/llvm.rb (hunk 2 above)
Comment (MikeMcQuaid) [inline Formula/llvm.rb]: `lib/dylib` works without the `.to_s` as these are strings as-is.

**Comment by MikeMcQuaid** [comment]
@ilovezfs I think so long as we build and install libc++ but want to keep using our own one: it's not their bug.

File: Formula/llvm.rb (hunk 2 above)
Comment (aw1621107) [inline Formula/llvm.rb]: Oh, ok. Fixed!

**Comment by ilovezfs** [comment]
Sorry but doing this looks like a mistake to me.

**Review by ilovezfs — CHANGES_REQUESTED** [review]
I don't think we should be making changes to the llvm formula to support superenv. This should be handled in superenv itself if it's necessary.

**Comment by MikeMcQuaid** [comment]
> I don't think we should be making changes to the llvm formula to support superenv. This should be handled in superenv itself if it's necessary.

@ilovezfs How? You can't add a library or include directory path without opening up everything else in that directory (which we don't want to do as we don't want to use the libc++ there).

**Comment by ilovezfs** [comment]
If you have "-L#{MacOS.sdk_path}/usr/lib -L/usr/local/opt/llvm/lib" it should prefer the brew libc++.

**Comment by MikeMcQuaid** [comment]
> If you have "-L#{MacOS.sdk_path}/usr/lib -L/usr/local/opt/llvm/lib" it should prefer the Xcode libc++.

:+1: to that approach. @aw1621107 it may be worth checking the ordering in superenv and seeing if that behaviour is as expected.

**Comment by aw1621107** [comment]
Is that behavior one we can count on to not change in the future?

**Comment by MikeMcQuaid** [comment]
Yes.

**Comment by aw1621107** [comment]
Alright, I'll see what I can do.

**Comment by JCount** [comment]
Pruning this due to the lack of a response and activity in general. Please comment if you would like to reopen and address the needed changes.

**Pull request closed without merging**